addons/web_editor/static/tests/html_field_tests.js

```diff
@@ -36,13 +134,469 @@ QUnit.module("WebEditor.HtmlField", ({ beforeEach }) => {
+                    this.save();
+                }, () => []);
+            },
+            save() {
+                const iconEl = document.createElement('span');
+                iconEl.classList.add('fa', 'fa-glass');
+                this.props.save(iconEl);
+                defMediaDialog.resolve();
+            },
+        });
+
+        wysiwyg.openMediaDialog();
+        await defMediaDialog;
+
+        let $editable = wysiwyg.$editable;
+
+        assert.strictEqual($editable.data('wysiwyg').getValue(),
+            '<p>t<span class="fa fa-glass"></span>to toto toto</p><p>tata</p>',
+            "should have the image in the dom");
+    });
+
+    QUnit.test("editor spellcheck is disabled on blur", async function (assert) {
+        const target = getFixture();
+
+        serverData.models['note.note'].records = [{
+            id: 1,
+            display_name: "first record",
+            header: "<p>  &nbsp;&nbsp;  <br>   </p>",
+            body: "<p>toto toto toto</p><p>tata</p>",
+        }];
+
+        await makeView({
+            type: "form",
+            resModel: "note.note",
+            resId: 1,
+            serverData,
+            arch: `<form><field name="body" widget="html" /></form>`,
+        });
+
+        const textarea = target.querySelector(".odoo-editor-editable");
+        textarea.focus();
+        assert.strictEqual(textarea.spellcheck, true, "spellcheck is enabled");
+        textarea.blur();
+        assert.strictEqual(
+            textarea.spellcheck,
+            false,
+            "spellcheck is disabled once the field has lost its focus"
+        );
+        textarea.focus();
+        assert.strictEqual(textarea.spellcheck, true, "spellcheck is re-enabled once the field is focused");
+    });
+
+    QUnit.module('cssReadonly');
+
+    QUnit.test('rendering with iframe for readonly mode', async function (assert) {
+        assert.expect(2);
+
+        legacyTestUtils.mock.patch(ajax, {
+            loadAsset: function (xmlId) {
+                if (xmlId === 'template.assets') {
+                    return Promise.resolve({
+                        cssLibs: [],
+                        cssContents: ['.o_in_iframe {background-color: red;}'],
+                        jsContents: ['window.odoo = {define: function(){}}; // inline asset'],
+                    });
+                }
+                if (xmlId === 'template.assets_all_style') {
+                    return Promise.resolve({
+                        cssLibs: $('link[href]:not([type="image/x-icon"])').map(function () {
+                            return $(this).attr('href');
+                        }).get(),
+                        cssContents: ['.o_in_iframe {background-color: red;}']
+                    });
+                }
+                if (xmlId === 'web_editor.wysiwyg_iframe_editor_assets') {
+                    return Promise.resolve({});
+                }
+                throw 'Wrong template';
+            },
+        });
```

Comment:
This one it's probably better to have it in the test itself than in the `beforeEach`, thanks ! 👍